hub/api/dataset.py

```diff
@@ -262,6 +262,39 @@ def load(
         except AgreementError as e:
             raise e from None
 
+    @staticmethod
+    def rename(
+        path: str,
+        name: str,
+        creds: Optional[dict] = None,
+        token: Optional[str] = None,
+    ) -> Dataset:
+        """Renames dataset at `path` to `name`.
+
+        Args:
+            path (str): The path to the dataset to be renamed.
+            name (str): Path to the dataset after renaming.
+            creds (dict, optional): A dictionary containing credentials used to access the dataset at the path.
+                This takes precedence over credentials present in the environment. Currently only works with s3 paths.
+                It supports 'aws_access_key_id', 'aws_secret_access_key', 'aws_session_token', 'endpoint_url' and 'region' as keys.
+            token (str, optional): Activeloop token, used for fetching credentials to the dataset at path if it is a Hub dataset. This is optional, tokens are normally autogenerated.
+
+        Returns:
+            Dataset object after renaming.
+
+        Raises:
+            DatasetHandlerError: If a Dataset does not exist at the given path or if new path is to a different directory.
+        """
+        if creds is None:
+            creds = {}
+
+        feature_report_path(path, "rename", {})
+
```

Comment:
we should check the path, before all of this.

Reply:
checking, should include multiple things, like checking that rename is happening from s3 to s3, gcs to gcs, hub to hub etc + for hub datasets we shouldn't allow renaming with username/xyz/ds